Tools/jit/_targets.py

```diff
@@ -455,13 +490,13 @@ def get_target(host: str) -> _COFF | _ELF | _MachO:
         return _ELF(host, alignment=8, args=args)
     if re.fullmatch(r"i686-pc-windows-msvc", host):
         args = ["-DPy_NO_ENABLE_SHARED"]
-        return _COFF(host, args=args, prefix="_")
```

Comment:
Is there some reason that we can't/don't want to do this for `aarch64`? https://llvm.org/docs/LangRef.html#calling-conventions seems to imply that it's supported.

Reply:
Yeah, but there's an annoying bug where Clang has an internal error if it's combined with `__attribute__((musttail))` (which is more important). Nice catch though... I'll add a comment explaining.

Reply:
```
fatal error: error in backend: failed to perform tail call elimination on a call site marked musttail
PLEASE submit a bug report to https://github.com/Homebrew/homebrew-core/issues and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.      Program arguments: /opt/homebrew/opt/llvm/bin/clang --target=aarch64-apple-darwin23.4.0 -DPy_BUILD_CORE_MODULE -D_DEBUG -D_JIT_OPCODE=_BINARY_OP -D_PyJIT_ACTIVE -D_Py_JIT -I. -I/Users/brandtbucher/cpython/Include -I/Users/brandtbucher/cpython/Include/internal -I/Users/brandtbucher/cpython/Include/internal/mimalloc -I/Users/brandtbucher/cpython/Python -O3 -c -fno-asynchronous-unwind-tables -fno-builtin -fno-plt -fno-stack-protector -std=c11 -Wno-unused-command-line-argument -o /private/var/folders/tx/dcr5klt14lgfdpwvmpt7lydc0000gn/T/tmpv05y4lth/_BINARY_OP.o /private/var/folders/tx/dcr5klt14lgfdpwvmpt7lydc0000gn/T/tmpv05y4lth/_BINARY_OP.ll
1.      Code generation
2.      Running pass 'Function Pass Manager' on module '/private/var/folders/tx/dcr5klt14lgfdpwvmpt7lydc0000gn/T/tmpv05y4lth/_BINARY_OP.ll'.
3.      Running pass 'AArch64 Instruction Selection' on function '@_JIT_ENTRY'
 #0 0x00000001175b9800 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0x4695800)
 #1 0x0000000113078d20 llvm::sys::CleanupOnSignal(unsigned long) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0x154d20)
 #2 0x00000001175ab068 (anonymous namespace)::CrashRecoveryContextImpl::HandleCrash(int, unsigned long) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0x4687068)
 #3 0x00000001175ab018 llvm::CrashRecoveryContext::HandleExit(int) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0x4687018)
 #4 0x00000001175b9054 llvm::sys::Process::Exit(int, bool) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0x4695054)
 #5 0x0000000100a23b8c LLVMErrorHandler(void*, char const*, bool) (/opt/homebrew/Cellar/llvm/18.1.4/bin/clang-18+0x100007b8c)
 #6 0x0000000112fbf68c llvm::report_fatal_error(llvm::Twine const&, bool) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0x9b68c)
 #7 0x00000001175ac1c0 llvm::report_fatal_error(llvm::StringRef, bool) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0x46881c0)
 #8 0x00000001154bf220 llvm::AArch64TargetLowering::LowerCall(llvm::TargetLowering::CallLoweringInfo&, llvm::SmallVectorImpl<llvm::SDValue>&) const (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0x259b220)
 #9 0x0000000113af5f3c llvm::TargetLowering::LowerCallTo(llvm::TargetLowering::CallLoweringInfo&) const (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0xbd1f3c)
#10 0x0000000113b1f350 llvm::SelectionDAGBuilder::lowerInvokable(llvm::TargetLowering::CallLoweringInfo&, llvm::BasicBlock const*) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0xbfb350)
#11 0x0000000113afcb38 llvm::SelectionDAGBuilder::LowerCallTo(llvm::CallBase const&, llvm::SDValue, bool, bool, llvm::BasicBlock const*) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0xbd8b38)
#12 0x0000000113acb89c llvm::SelectionDAGBuilder::visitCall(llvm::CallInst const&) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0xba789c)
#13 0x0000000113bbdfa0 llvm::SelectionDAGISel::SelectBasicBlock(llvm::ilist_iterator_w_bits<llvm::ilist_detail::node_options<llvm::Instruction, false, false, void, true>, false, true>, llvm::ilist_iterator_w_bits<llvm::ilist_detail::node_options<llvm::Instruction, false, false, void, true>, false, true>, bool&) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0xc99fa0)
#14 0x0000000113bbcbd8 llvm::SelectionDAGISel::SelectAllBasicBlocks(llvm::Function const&) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0xc98bd8)
#15 0x0000000113bb9ec0 llvm::SelectionDAGISel::runOnMachineFunction(llvm::MachineFunction&) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0xc95ec0)
#16 0x000000011357c694 llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0x658694)
#17 0x0000000113237920 llvm::FPPassManager::runOnModule(llvm::Module&) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0x313920)
#18 0x000000011322bc4c llvm::legacy::PassManagerImpl::run(llvm::Module&) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0x307c4c)
#19 0x0000000106e5ef70 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem>, std::__1::unique_ptr<llvm::raw_pwrite_stream, std::__1::default_delete<llvm::raw_pwrite_stream>>, clang::BackendConsumer*) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libclang-cpp.dylib+0x1596f70)
#20 0x0000000107209c80 clang::CodeGenAction::ExecuteAction() (/opt/homebrew/Cellar/llvm/18.1.4/lib/libclang-cpp.dylib+0x1941c80)
#21 0x0000000107965acc clang::FrontendAction::Execute() (/opt/homebrew/Cellar/llvm/18.1.4/lib/libclang-cpp.dylib+0x209dacc)
#22 0x00000001078e87d0 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libclang-cpp.dylib+0x20207d0)
#23 0x00000001079b88e8 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libclang-cpp.dylib+0x20f08e8)
#24 0x0000000100a22b28 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/opt/homebrew/Cellar/llvm/18.1.4/bin/clang-18+0x100006b28)
#25 0x0000000100a21218 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&, llvm::ToolContext const&) (/opt/homebrew/Cellar/llvm/18.1.4/bin/clang-18+0x100005218)
#26 0x0000000107566b04 void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<std::__1::optional<llvm::StringRef>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>*, bool*) const::$_0>(long) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libclang-cpp.dylib+0x1c9eb04)
#27 0x0000000112fb0460 llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib+0x8c460)
#28 0x0000000107566300 clang::driver::CC1Command::Execute(llvm::ArrayRef<std::__1::optional<llvm::StringRef>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>*, bool*) const (/opt/homebrew/Cellar/llvm/18.1.4/lib/libclang-cpp.dylib+0x1c9e300)
#29 0x0000000107526f68 clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&, bool) const (/opt/homebrew/Cellar/llvm/18.1.4/lib/libclang-cpp.dylib+0x1c5ef68)
#30 0x0000000107547bd8 clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::__1::pair<int, clang::driver::Command const*>>&) (/opt/homebrew/Cellar/llvm/18.1.4/lib/libclang-cpp.dylib+0x1c7fbd8)
#31 0x0000000100a1fd44 clang_main(int, char**, llvm::ToolContext const&) (/opt/homebrew/Cellar/llvm/18.1.4/bin/clang-18+0x100003d44)
#32 0x0000000100a2ce38 main (/opt/homebrew/Cellar/llvm/18.1.4/bin/clang-18+0x100010e38)
#33 0x000000019681a0e0 
clang: error: clang frontend command failed with exit code 70 (use -v to see invocation)
Homebrew clang version 18.1.4
Target: aarch64-apple-darwin23.4.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm/bin
```